packages/evm/contracts/registry/GuardiansRegistry.sol

```diff
@@ -168,78 +169,82 @@ contract GuardiansRegistry is ControlledRecoverable, IGuardiansRegistry, IERC900
-        _stake(msg.sender, _to, _amount, _data);
+    function deactivate(address _guardian, uint256 _amount) external authenticate(_guardian) {
+        _deactivate(_guardian, _amount);
     }
 
     /**
-    * @notice Unstake `@tokenAmount(self.token(), _amount)` for `_to` from the Protocol
-    * @param _amount Amount of tokens to be unstaked
+    * @notice Stake and activate `@tokenAmount(self.token(), _amount)` for `_guardian`
+    * @param _guardian Address of the guardian staking and activating tokens for
+    * @param _amount Amount of tokens to be staked and activated
     * @param _data Optional data is never used by this function, only logged
     */
-    function unstake(uint256 _amount, bytes calldata _data) external {
-        _unstake(msg.sender, _amount, _data);
-    }
+    function stakeAndActivate(address _guardian, uint256 _amount, bytes calldata _data) external {
+        // Make sure the sender is the guardian or someone allowed by the guardian, and that the activator is whitelisted
+        bool isActivatorAllowed = _isSignatureValid(_guardian) || _isActivatorWhitelisted(msg.sender);
+        require(isActivatorAllowed, ERROR_ACTIVATOR_NOT_ALLOWED);
 
-    /**
-    * @dev Callback of approveAndCall, allows staking directly with a transaction to the token contract.
-    * @param _from Address making the transfer
-    * @param _amount Amount of tokens to transfer
-    * @param _token Address of the token
-    * @param _data Optional data that can be used to request the activation of the transferred tokens
-    */
-    function receiveApproval(address _from, uint256 _amount, address _token, bytes calldata _data) external {
-        require(msg.sender == _token && _token == address(guardiansToken), ERROR_TOKEN_APPROVE_NOT_ALLOWED);
-        _stake(_from, _from, _amount, _data);
+        _stake(_guardian, _amount, _data);
+        _activate(_guardian, _amount);
     }
 
     /**
-    * @notice Lock `@tokenAmount(self.token(), _amount)` of sender's active balance
+    * @notice Lock `@tokenAmount(self.token(), _amount)` of `_guardian`'s active balance
+    * @param _guardian Address of the guardian locking the activation for
     * @param _lockManager Address of the lock manager that will control the lock
     * @param _amount Amount of active tokens to be locked
     */
-    function lockActivation(address _lockManager, uint256 _amount) external {
-        _lockActivation(msg.sender, _lockManager, _amount);
+    function lockActivation(address _guardian, address _lockManager, uint256 _amount) external {
+        // Make sure the sender is the guardian, someone allowed by the guardian, or the lock manager itself
+        bool isLockManagerAllowed = msg.sender == _lockManager || _isSignatureValid(_guardian);
+        // Make sure that the given lock manager is whitelisted or that any entity actually is
+        bool isLockManagerWhitelisted = _isLockManagerWhitelisted(_lockManager) || _isLockManagerWhitelisted(ANY_ENTITY);
```

Comment:
⚠️ Given this change, the `ANY_ENTITY` is pretty useless :). It would mean any entity would be allowed to select any guardian and lock an arbitrary amount of tokens.

Reply:
Yea, I agree with that, let's remove this one